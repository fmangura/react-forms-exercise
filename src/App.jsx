import { useState } from 'react'
import './App.css'
import BoxList from './components/color-box/BoxList'
import TodoList from './components/Todo/TodoList'

function App() {
  const [count, setCount] = useState(0)

  return (
    <>
      <TodoList />
      <BoxList />
    </>
  )
}

export default App
