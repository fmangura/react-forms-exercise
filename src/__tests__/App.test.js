import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom'
import App from '../App.jsx';

it('renders without crashing', function() {
    render(<App />);
})

it('matches snapshot', function() {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
})

it('check contents', function() {
    const { getByText } = render(<App />);
    expect(getByText('To Do:', { exact: false })).toBeInTheDocument();
    expect(getByText('Boxes:', { exact: false })).toBeInTheDocument();
})