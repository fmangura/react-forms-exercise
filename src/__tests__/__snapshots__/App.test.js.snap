// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <form>
        <label
          for="Todo"
        >
          What to do?
        </label>
        <input
          id="Todo"
          name="Todo"
          style="width: 500px;"
          type="text"
        />
        <br />
        <button>
          Submit!
        </button>
      </form>
    </div>
  </div>
  <h1>
    To Do:
  </h1>
  <hr />
  <div>
    <div>
      <form>
        <label
          for="color"
        >
          Color: 
        </label>
        <input
          id="color"
          name="color"
          placeholder="Color"
          type="text"
          value=" "
        />
        <label
          for="width"
        >
          Width px: 
        </label>
        <input
          id="width"
          name="width"
          placeholder="Width px"
          type="integer"
          value=" "
        />
        <label
          for="height"
        >
          Height px: 
        </label>
        <input
          id="height"
          name="height"
          placeholder="Height px"
          type="integer"
          value=" "
        />
        <button>
          Add Box
        </button>
      </form>
    </div>
    <h1>
      Boxes:
    </h1>
    <div
      class="Boxes"
    />
  </div>
</DocumentFragment>
`;
